standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionInfo.java

```diff
@@ -87,6 +89,21 @@ public CompactionInfo(String dbname, String tableName, String partName, Compacti
   }
   CompactionInfo() {}
 
+  public String getProperty(String key) {
+    if (propertiesMap == null) {
+      propertiesMap =new StringableMap(properties);
+    }
+    return propertiesMap.get(key);
+  }
+
+  public void setProperty(String key, String value) {
+    if (propertiesMap == null) {
```

Comment:
why not to initialize map in constructor or something?

Reply:
The lazy loading is to save some resources. I believe that in most of the cases when a _CompactionInfo_ instance is loaded, the properties are not used at all.